ebos/eclmpiserializer.hh

```diff
@@ -48,6 +48,32 @@ public:
             Mpi::unpack(const_cast<T&>(data), m_buffer, m_position, m_comm);
     }
 
+    template<class T>
+    void vector(const std::vector<T>& datac)
```

Comment:
Why const in the first place if you cast it away anyway

Reply:
well, the "problem" is the the single serialize/deserialize approach in combination with the underlying serializer. I need to cast somewhere since packSize/pack takes consts and unpack take non-const and the compiler  is not smart enough to choose the correct template instance for packSize/pack if we pass a non-const reference. I earlier chose to eat by const ref in operator() to limit casting to one place.

strictly speaking i could only cast for operation == Operation::UNPACK here, but I kept it as-is for simplicity. i can do the limited casting if you prefer.

Reply:
Actually, come to think of it I cannot do this. The serializeOp member in the data types are non-const since they do both serialization and deserialization. What I can do, is not eat the parameter as const in the first place. And that makes a lot of sense, I got hung up in the operator() approach.